qiita_db/sql_connection.py

```diff
@@ -402,9 +430,16 @@ def execute_fetchall(self, sql, sql_args=None):
         """
         with self._sql_executor(sql, sql_args) as pgcursor:
             result = pgcursor.fetchall()
-        return result
+            if return_types:
```

Comment:
Looking again, they do need to be within the `with` for access to the pgcursor. Even with leaky variables, it makes more sense visually when following what's going on.